reflex/app.py

```diff
@@ -404,10 +404,13 @@ def _check_routes_conflict(self, new_route: str):
 
     def add_custom_404_page(
         self,
-        component,
+        component=None,
```

Comment:
nit: will be nice to have  annotations for all params for consistency and readability